plutus-core/untyped-plutus-core/src/Language/UntypedPlutusCore/Evaluation/Machine/Cek.hs

```diff
@@ -40,496 +37,29 @@ module Language.UntypedPlutusCore.Evaluation.Machine.Cek
     )
 where
 
-import           ErrorCode
 import           PlutusPrelude
 
 import           Language.UntypedPlutusCore.Core
-import           Language.UntypedPlutusCore.Subst
+import           Language.UntypedPlutusCore.Evaluation.Machine.Cek.ExBudgetMode
+import           Language.UntypedPlutusCore.Evaluation.Machine.Cek.Internal
 
 import           Language.PlutusCore.Constant
-import           Language.PlutusCore.Evaluation.Machine.ExBudgeting
+import           Language.PlutusCore.Evaluation.Machine.ExBudget
 import           Language.PlutusCore.Evaluation.Machine.ExMemory
 import           Language.PlutusCore.Evaluation.Machine.Exception
-import           Language.PlutusCore.Evaluation.Result
 import           Language.PlutusCore.Name
 import           Language.PlutusCore.Pretty
 import           Language.PlutusCore.Universe
 
-import           Control.Lens                                       (AReview)
-import           Control.Monad.Except
-import           Control.Monad.Morph
-import           Control.Monad.Reader
-import           Control.Monad.ST
-import           Control.Monad.State.Strict
-import           Data.Array
-import           Data.DList                                         (DList)
-import qualified Data.DList                                         as DList
-import           Data.HashMap.Monoidal
-import           Data.STRef
-import           Data.Text.Prettyprint.Doc
-
-{- Note [Scoping]
-   The CEK machine does not rely on the global uniqueness condition, so the renamer pass is not a
-   prerequisite. The CEK machine correctly handles name shadowing.
--}
```

Comment:
I've separated the implementation of the CEK machine (`Language.UntypedPlutusCore.Evaluation.Machine.Cek.Internal`) from the available budgeting modes and their logic (`Language.UntypedPlutusCore.Evaluation.Machine.Cek.ExBudgetMode`) from the final API to the CEK machine (`Language.UntypedPlutusCore.Evaluation.Machine.Cek`).